tests_consume/test_via_rlp.py

```diff
@@ -0,0 +1,244 @@
+"""
+Test module that defines a test to execute a fixture against an EVM blocktest-like command.
+"""
+import json
+import tempfile
+from pathlib import Path
+from typing import List, Literal, Union
+
+import pytest
+import requests
+from hive.client import Client
+from hive.testing import HiveTest, HiveTestResult, HiveTestSuite
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ethereum_test_tools.common.types import Fixture
+from pytest_plugins.consume_via_rlp.consume_via_rlp import TestCase
+from pytest_plugins.consume_via_rlp.network_ruleset_hive import ruleset
+
+
+@pytest.fixture(scope="function")
+def temp_dir() -> tempfile.TemporaryDirectory:
+    """
+    Return a temporary directory to write the genesis.json and block RLP files to.
+    """
+    return tempfile.TemporaryDirectory()
+
+
+@pytest.fixture(scope="function")
+def test_case_fixture(test_case: TestCase) -> Fixture:
+    """
+    The test fixture as a dictionary.
+    """
+    assert test_case.fixture is not None
+    return test_case.fixture
+
+
+@pytest.fixture(scope="function")
+def expected_hash(test_case_fixture: Fixture) -> str:
+    """
+    The hash defined in the test fixture's last block header.
+    """
+    return test_case_fixture.blocks[-1]["blockHeader"]["hash"]
+
+
+@pytest.fixture(scope="function")
+def expected_state_root(test_case: TestCase) -> str:
+    """
+    The state root defined in the test fixture's last block header.
+    """
+    return test_case.fixture.blocks[-1]["blockHeader"]["stateRoot"]
+
+
+# @pytest.fixture
+# def execution_client(request):
+#    execution_clients = request.cw
+#    # assert len(execution_client) == 1
+#    return execution_clients[0]
+
+
+@pytest.fixture
+def hive_test(request, test_suite: HiveTestSuite):
+    test_parameter_string = request.node.nodeid.split("[")[-1].rstrip("]")
+    test: HiveTest = test_suite.start_test(
+        name=test_parameter_string, description="TODO: This should come from the '_info' field."
+    )
+    yield test
+    test_result: HiveTestResult
+    try:
+        if hasattr(request.node, "rep_call"):
+            test_passed = request.node.rep_call.passed
```

Comment:
It seems like `node` never contains `rep_call` for me.

Reply:
Improved in [30e2242](https://github.com/ethereum/execution-spec-tests/pull/339/commits/30e2242c903bdf6510272177be25d71865802816).

Probably needs further refinement for xfail/skip though.